### PR TITLE
[FEAT] support cos/tan/asin/acos function

### PR DESCRIPTION
**Pull request by gaoxinge**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/runtime/function/tan.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+type tanFunc struct{}
+
+// Apply call the current function.
+func (s tanFunc) Apply(ctx context.Context, inputs ...proto.Valuer) (proto.Value, error) {
+	param, err := inputs[0].Value(ctx)
+	if param == nil || err != nil {
+		return nil, errors.WithStack(err)
+	}
```
Comment (jjeffcaii) [inline pkg/runtime/function/tan.go:49]: image

When `param` here is nil, should return `nil,nil`.

File: pkg/runtime/function/acos.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+type acosFunc struct{}
+
+// Apply call the current function.
+func (s acosFunc) Apply(ctx context.Context, inputs ...proto.Valuer) (proto.Value, error) {
+	param, err := inputs[0].Value(ctx)
+	if param == nil || err != nil {
+		return nil, errors.WithStack(err)
+	}
```
Comment (jjeffcaii) [inline pkg/runtime/function/acos.go:49]: image
When `param` here is nil, should return `nil,nil`.

PS: please check the rest of functions in this PR, there're same problem.

**Review by jjeffcaii — COMMENTED** [review]
pls fix

File: pkg/runtime/function/acos.go (hunk 2 above)
Comment (gaoxinge) [reply pkg/runtime/function/acos.go:49]: We consider two cases:

- If `err != nil`, the function will return `nil, errors.WithStack(err)`  as before.
- If `param == nil && err == nil`, the function will return `nil, nil` as before too.  This is because `errors.WithStack` return `nil`, if input is `nil`.

withstack

I add more test cases for these math functions, which includes the `nil` input.

Comment (jjeffcaii) [reply pkg/runtime/function/acos.go:49]: Oops... it seems right. 🤣

**Review by dongzl — APPROVED** [review]
